File: backend/financier/src/main/java/com/welpeth/kakebo/financier/domain/account/service/AccountService.java
package com.welpeth.kakebo.financier.domain.account.service;

import com.welpeth.kakebo.financier.domain.account.dto.CreateAccountRequest;
import com.welpeth.kakebo.financier.domain.account.dto.UpdateAccountRequest;import com.welpeth.kakebo.financier.domain.account.entity.Account;
import com.welpeth.kakebo.financier.domain.account.repository.AccountRepository;

import java.util.List;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AccountService {

  private final AccountRepository repository;

  public Account create(CreateAccountRequest request) {
    Account account = new Account();
    account.setId(UUID.randomUUID());
    account.setName(request.name());
    account.setBalance(request.balance());
    account.setIsActive(request.isActive());
    account.setType(request.type());

    repository.save(account);
    return account;
  }

  public Account get(UUID id) {
    return repository.getReferenceById(id);
  }

  public void update(UpdateAccountRequest request) {
    repository.update(request);
  }

  public void delete(UUID id) {
    repository.deleteById(id);
  }

  public List<Account> getList() {
    return repository.findAll();
  }
}

File: frontend/src/services/account.service.ts
import api from '@/lib/api'
import type { Account, CreateAccountRequest, UpdateAccountRequest } from '@/models/models'

export const accountService = {
  async findAll(): Promise<Account[]> {
    const { data } = await api.get<Account[]>('/account/list')
    return data
  },

  async findById(id: string): Promise<Account> {
    const { data } = await api.get<Account>(`/account/${id}`)
    return data
  },

  async create(payload: CreateAccountRequest): Promise<Account> {
    const { data } = await api.post<Account>('/account', payload)
    return data
  },

  async update(payload: UpdateAccountRequest): Promise<void> {
    await api.patch('/account', payload)
  },

  async remove(id: string): Promise<void> {
    await api.delete(`/account/${id}`)
  },
}
